doc/source/whatsnew/v1.0.0.rst

```diff
@@ -652,6 +652,7 @@ Datetimelike
 - Bug in :meth:`Series.var` failing to raise ``TypeError`` when called with ``timedelta64[ns]`` dtype (:issue:`28289`)
 - Bug in :meth:`DatetimeIndex.strftime` and :meth:`Series.dt.strftime` where ``NaT`` was converted to the string ``'NaT'`` instead of ``np.nan`` (:issue:`29578`)
 - Bug in :attr:`Timestamp.resolution` being a property instead of a class attribute (:issue:`29910`)
+- Bug in :func:`pandas.core.tools.datetimes._attempt_YYYYMMDD` to_datetime() when called with ``None`` raising ``TypeError`` instead of returning ``NaT``. After changes it will return ``NaT`` (:issue:`30011`)
```

Comment:
don't put an internal function, rather just `~pandas.to_datetime` is enough; also remove the 'After changes ....' is redundant with your prior comment.